nltk/corpus/reader/extended_omw.py

```diff
@@ -0,0 +1,141 @@
+# Natural Language Toolkit: Extened Open Multilingual WordNet Reader
+#
+# Copyright (C) 2001-2021 NLTK Project
+# Author: Freda Shi <freda@ttic.edu>
+
+# URL: <https://www.nltk.org/>
+# For license information, see LICENSE.TXT
+
+"""
+An NLTK interface for Extended Open Multilingual WordNet
+
+Extended Open Multilingual WordNet automatically maps WordNet synsets
+to multiple languages with data extracted from Wikitionary and CLDR. 
+
+Currently ignoring all languages marked with "*" in the released corpus. 
+All synsets, whether manually annotated or automatically extracted, are 
+treated equally. 
+
+For details about WordNet, see:
+https://wordnet.princeton.edu/
+
+For details about Open Multilingual WordNet, see:
+http://compling.hss.ntu.edu.sg/omw/
+
+For details about Extended Open Multilingual WordNet, see:
+http://compling.hss.ntu.edu.sg/omw/summx.html
+"""
+
+import os
+from collections import defaultdict
+from IPython.core.pylabtools import configure_inline_support
+
+from traitlets.traitlets import default
+from nltk.corpus.reader.wordnet import WordNetCorpusReader
+
+
+class ExtendedOpenMultilingualWordNetCorpusReader(WordNetCorpusReader):
+    def __init__(self, root, omw_reader, exomw_reader):
+        super().__init__(root, omw_reader)
+        self._exomw_reader = exomw_reader
+        self.exomw_provenances = self.exomw_prov()
+
+    def omw_prov(self):
+        """Return a provenance dictionary of the languages in OMW"""
+        provdict = defaultdict(list)
+        provdict["eng"] = list()
+        fileids = self._omw_reader.fileids()
+        for fileid in fileids:
+            prov, langfile = os.path.split(fileid)
+            file_name, file_extension = os.path.splitext(langfile)
+            if file_extension == ".tab":
+                lang = file_name.split("-")[-1]
+                provdict[lang].append(prov)
+        return provdict
+
+    def exomw_prov(self):
+        """Return a provenance dictionary of the languages in Extended OMW"""
+        provdict = defaultdict(list)
+        provdict["eng"] = ""
+        fileids = self._exomw_reader.fileids()
+        for fileid in fileids:
+            prov, langfile = os.path.split(fileid)
+            file_name, file_extension = os.path.splitext(langfile)
+            if file_extension == ".tab":
+                lang = file_name.split("-")[-1]
+                # only use wordnet English data -- should be examined.
+                if lang == "eng": 
+                    continue
+                provdict[lang].append(prov)
+        return provdict
+
+    def _load_lang_data(self, lang):
+        """Load the wordnet data to self._lang_data"""
+        if lang in self._lang_data.keys():
+            return
+        # load open multilingual wordnet
+        for prov in self.provenances[lang]:
+            with self._omw_reader.open(f"{prov}/wn-data-{lang}.tab") as fp:
+                self.custom_lemmas(fp, lang)
+                fp.close()
```

Comment:
The `with` keyword already automatically closes any files whenever the `with` scope is exited, so this line is unnecessary.